src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -727,17 +748,18 @@ Real SpatialPooler::avgConnectedSpanForColumnND_(UInt column) const {
   NTA_ASSERT(column < numColumns_);
   
   const UInt numDimensions = inputDimensions_.size();
-  const vector<UInt> connectedSparse = connectedSynapses_.getSparseRow(column);
-  if (connectedSparse.empty()) {
-    return 0;
-  }
+
+  vector<UInt> connectedDense( numInputs_, 0 );
+  getConnectedSynapses( column, connectedDense.data() );
```

Comment:
can we get this in sparse format?

Reply:
Method `SP.getConnectedSynapses()` is public facing API...